web/web.go

```diff
@@ -0,0 +1,771 @@
+	}
+	if srcCol.Ignored.AutoIncrement {
+		issues = append(issues, internal.AutoIncrement)
+	}
+	if len(issues) > 0 {
+		app.conv.Issues[srcTableName][srcCol.Name] = issues
+	}
+	ty.IsArray = len(srcCol.Type.ArrayBounds) == 1
+	tempColDef := sp.ColDefs[newColName]
+	tempColDef.T = ty
+	sp.ColDefs[newColName] = tempColDef
+	app.conv.SpSchema[table] = sp
+}
+
+func updateNotNull(notNullChange, table, newColName string) {
+	sp := app.conv.SpSchema[table]
+	switch notNullChange {
+	case "ADDED":
+		spColDef := sp.ColDefs[newColName]
+		spColDef.NotNull = true
+		sp.ColDefs[newColName] = spColDef
+	case "REMOVED":
+		spColDef := sp.ColDefs[newColName]
+		spColDef.NotNull = false
+		sp.ColDefs[newColName] = spColDef
+	default:
+		// We skip this.
+	}
+	app.conv.SpSchema[table] = sp
+}
+
+// Actions to be performed on a column.
+// (1) Removed: true/false
+// (2) Rename: New name or empty string
+// (3) PK: "ADDED", "REMOVED" or ""
+// (4) NotNull: "ADDED", "REMOVED" or ""
+// (5) ToType: New type or empty string
+type updateCol struct {
+	Removed bool   `json:"Removed"`
+	Rename  string `json:"Rename"`
+	PK      string `json:"PK"`
+	NotNull string `json:"NotNull"`
+	ToType  string `json:"ToType"`
+}
+type updateTable struct {
+	UpdateCols map[string]updateCol `json:"UpdateCols"`
+}
+
+func updateTableSchema(w http.ResponseWriter, r *http.Request) {
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+		return
+	}
+	var t updateTable
+	table := r.FormValue("table")
+	err = json.Unmarshal(reqBody, &t)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
+		return
+	}
+	srcTableName := app.conv.ToSource[table].Name
+	for colName, v := range t.UpdateCols {
+		if v.Removed {
+			removeColumn(table, colName, srcTableName)
+			continue
+		}
+		newColName := colName
+		if v.Rename != "" && v.Rename != colName {
+			renameColumn(v.Rename, table, colName, srcTableName)
+			newColName = v.Rename
+		}
+		if v.PK != "" {
+			pkChanged(v.PK, table, newColName)
+		}
+		if v.ToType != "" {
+			updateType(v.ToType, table, newColName, srcTableName, w)
+		}
+		if v.NotNull != "" {
+			updateNotNull(v.NotNull, table, newColName)
+		}
+	}
+	app.conv.AddPrimaryKeys()
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(app.conv)
+}
+
+func rateSchema(cols, warnings int64, missingPKey bool) string {
+	switch {
+	case cols == 0:
+		return "GRAY"
+	case warnings == 0 && !missingPKey:
+		return "GREEN"
+	case warnings == 0 && missingPKey:
+		return "BLUE"
+	case good(cols, warnings) && !missingPKey:
+		return "BLUE"
+	case good(cols, warnings) && missingPKey:
+		return "BLUE"
+	case ok(cols, warnings) && !missingPKey:
+		return "YELLOW"
+	case ok(cols, warnings) && missingPKey:
+		return "YELLOW"
+	case !missingPKey:
+		return "RED"
+	default:
+		return "RED"
+	}
+}
+func good(total, badCount int64) bool {
```

Comment:
Probably just defined good and ok as local functions in rateSchema, since that is where they are used.

i.e. good := func(total, badCount int64) bool { return badCount < total/20 }

Reply:
made good and ok local function